### PR TITLE
Convert model inference test from pytest to unittest

### PR DESCRIPTION
**Pull request by namgyu-youn**
**Summary:**
Converts TorchAO model (`Transformer`) inference test from Pytest into Unittest. Annotations are added for better readability.
- Related to: #1621 

cc @osbm

### CODE REVIEW

File: test/test_ao_models.py [hunk 1]
```diff
@@ -12,23 +13,57 @@
...
+        """Test model inference across different devices, batch sizes, and training modes."""
+        # Define test parameters
+        devices = _AVAILABLE_DEVICES
+        batch_sizes = [1, 4]
+        training_modes = [True, False]
+
+        # Iterate through all parameter combinations
+        for device in devices:
+            for batch_size in batch_sizes:
+                for is_training in training_modes:
```
Comment (jerryzh168) [inline test/test_ao_models.py]: nit: can change these to `common_utils.parametrize` like: https://github.com/pytorch/ao/blob/0935f66d45acd30dc41bd955d042a1fee70cffc6/torchao/testing/utils.py#L186-L187 and add a instantiation in the end: https://github.com/pytorch/ao/blob/0935f66d45acd30dc41bd955d042a1fee70cffc6/torchao/testing/utils.py#L425

Comment (namgyu-youn) [reply test/test_ao_models.py]: Love these decorators, thanks!

File: test/test_ao_models.py [hunk 2]
```diff
@@ -3,32 +3,59 @@
...
+import unittest
+
 import torch
 
 from torchao._models.llama.model import Transformer
+from torchao.testing import common_utils
 
 _AVAILABLE_DEVICES = ["cpu"] + (["cuda"] if torch.cuda.is_available() else [])
+_BATCH_SIZES = [1, 4]
+_TRAINING_MODES = [True, False]
```
Comment (jerryzh168) [inline test/test_ao_models.py]: nit: please feel free to inline these

File: test/test_ao_models.py [hunk 3]
```diff
@@ -3,32 +3,59 @@
...
+
+# Define test parameters
+COMMON_DEVICES = common_utils.parametrize("device", _AVAILABLE_DEVICES)
+COMMON_DTYPES = common_utils.parametrize("dtype", [torch.float32, torch.bfloat16])
```
Comment (jerryzh168) [inline test/test_ao_models.py]: we don't need to define these here, can just put these in the function itself

File: test/test_ao_models.py [hunk 4]
```diff
@@ -3,32 +3,53 @@
...
 # LICENSE file in the root directory of this source tree.
-import pytest
+import unittest
+
 import torch
 
 from torchao._models.llama.model import Transformer
-
-_AVAILABLE_DEVICES = ["cpu"] + (["cuda"] if torch.cuda.is_available() else [])
+from torchao.testing import common_utils
```
Comment (jerryzh168) [inline test/test_ao_models.py]: the import is not correct, please run the tests locally first to make sure the test runs

Comment (namgyu-youn) [reply test/test_ao_models.py]: Sorry for my carelessness. I fixed it and tested it locally.

**Comment by jerryzh168** [comment]
we will need to use the merge button to merge in torchao

**Review by jerryzh168 — APPROVED** [review]

**Comment by namgyu-youn** [comment]
> we will need to use the merge button to merge in torchao

Like this because there is no "merged" label, thanks. Also, could you check https://github.com/pytorch/ao/pull/2660 ? It replaces `torch.norm` with `torch.linalg.vector_norm` for PyTorch future update.

**Comment by namgyu-youn** [comment]
@jerryzh168 gentle ping for the reminder :)
